agent/src/beerocks/slave/beerocks_slave_main.cpp

```diff
@@ -299,13 +306,14 @@ static int run_beerocks_slave(beerocks::config_file::sConfigSlave &beerocks_slav
         beerocks_slave_conf.temp_path + "pid/" + base_slave_name; // for file touching
 
     std::set<std::string> slave_ap_ifaces;
-    for (int slave_num = 0; slave_num < beerocks::IRE_MAX_SLAVES; slave_num++) {
-        if (!beerocks_slave_conf.hostap_iface[slave_num].empty()) {
-            slave_ap_ifaces.insert(beerocks_slave_conf.hostap_iface[slave_num]);
+    for (auto &elm : interfaces_map) {
+        if (!elm.second.empty()) {
+            slave_ap_ifaces.insert(elm.second);
```

Comment:
Why do we need another data structure?
If we already create a map with the interfaces (and potentially all other related data), why not use everywhere?

Reply:
the BH manager needs the list of all interfaces, while the interfaces_map is an agent/radio-->iface map.
we can provide the map to the backhaul and let it take care of extracting the information but I think that changing the CTOR signature will hide the actual "requirements" of the BH manager.

I leave the call to you.

Reply:
One map to rule them all!
Interface map is basically a radio map, which should have the VAPs and all other related data as Vitaly suggests. But since it is a bigger change I'm not sure.. @vitalybu ?

Reply:
I can change the interface_map to be radio_map and expand it to contain the rest of the param values as part of the PR that moves the rest of the params to the prplmesh config file.
Unless you think it must be part of this PR, and I will need to change the backhaul_manager implementation accordingly.

Reply:
talked with @vitalybu and the suggested change will be handled as part of the PR to move rest of params to the prplmesh config file